pkg/controller/workload/job/job_controller.go

```diff
@@ -173,6 +238,10 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		if jobFinished {
 			return ctrl.Result{}, nil
 		}
+		if parentWorkload(&job) != "" {
+			log.V(3).Info("The parent workload '%s' does not exist", parentWorkload(&job))
```

Comment:
Not sure if we should log anything here, or maybe we should give more information.

Is there already a log line explaining that the job was suspended? If it exists, maybe we should add a field for the missing Workload there. If not, we can say here that the Workload was suspended.

Also, we use structured logging, not `%s`.

Reply:
This remark led me to identify a bug - a job without a parent workload wound't be suspended. There is an integration test aiming to test that, but it was starting from already suspended job. 

Now fixed the test and implementation. I also drop this log like, as I added another one to mention that the parent workload is missing just after trying to get it by the client.